compiler/prelude/types.go

```diff
@@ -617,14 +617,15 @@ var $mapType = function(key, elem) {
   return typ;
 };
 var $makeMap = function(keyForFunc, entries) {
-  var m = {};
-  for (var i = 0; i < entries.length; i++) {
-    var e = entries[i];
-    m[keyForFunc(e.k)] = e;
-  }
-  return m;
+ //console.log("$makeMap input", keyForFunc, entries) 
+ var m = new Map();
+ for (var i = 0; i < entries.length; i++) {
+   var e = entries[i];
+   m.set(keyForFunc(e.k), e.v);
+ }
+ //console.log("$makeMap output", m)
+ return m;
```

Comment:
Nit: Indentation seems to have been changed here, which makes discerning the meaningful changes a bit harder.